utils/etherspot.js

```diff
@@ -42,10 +43,21 @@ const wstEthAddress = "0x5979D7b546E38E414F7E9822514be443A4800529";
 const CamelotNFTPositionManagerAddress =
   "0x00c7f3082833e796A5b3e4Bd59f6642FF44DCD15";
 
-export async function investByAAWallet(investmentAmount, chosenToken) {
+export async function investByAAWallet(
+  investmentAmount,
+  chosenToken,
+  walletClient,
+) {
   console.log("Investing by AA Wallet...");
   console.log("chosenToken", chosenToken);
-  const portfolioHelper = await getPortfolioHelper("AllWeatherPortfolio");
+  const mappedProvider = new Web3WalletProvider(walletClient);
+  // await mappedProvider.refresh();
```

Comment:
`refresh()` would raise this error `POST https://arb1.arbitrum.io/rpc net::ERR_FAILED 400 (Bad Request)`

Should I use etherspot's RPC or?